cs/src/core/Allocator/AllocatorBase.cs

```diff
@@ -1471,9 +1473,14 @@ private void AsyncGetFromDiskCallback(uint errorCode, uint numBytes, NativeOverl
                     // We have the complete record.
                     if (RetrievedFullRecord(record, ref ctx))
                     {
-                        if (comparer.Equals(ref ctx.request_key.Get(), ref GetContextRecordKey(ref ctx)))
+                        // If the keys are same, then I/O is complete. For PSFs, we may be querying on an address instead of a key;
+                        // in this case, ctx.request_key is null for the primary FKV but we know the address is the one we want.
+                        var isEqualKey = ctx.request_key is null 
+                                        || (this.PsfKeyAccessor is null 
+                                            ? comparer.Equals(ref ctx.request_key.Get(), ref GetContextRecordKey(ref ctx))
+                                            : this.PsfKeyAccessor.EqualsAtRecordAddress(ref KeyPointer<Key>.CastFromKeyRef(ref ctx.request_key.Get()), (long)record));
```

Comment:
This (this.PsfKeyAccessor.EqualsAtRecordAddress) is for the secondary FKV, correct? We can just use this code as a custom comparer.